parser.js

```diff
@@ -1,273 +1,51 @@
+    function render(html, context) {
+        var replaceText = {
+            closeTag: context.closeTag && context.closeTag.name ? context.closeTag.name : "",
+            openTag: context.openTag && context.openTag.name ? context.openTag.name : "",
+            attribute: context.attribute && context.attribute.name && context.attribute.name.value ? context.attribute.name.value : "",
+            cssValue: context.cssValue && context.cssValue.value ? context.cssValue.value : "",
+            cssSelector: context.cssSelector && context.cssSelector.selector ?  context.cssSelector.selector : "",
+            cssProperty: context.cssProperty && context.cssProperty.property ? context.cssProperty.property : "",
+            cssKeyword: context.cssKeyword && context.cssKeyword.value ? context.cssKeyword.value : "",
+            name: context.name || "",
+            error: context.error && context.error.msg ? context.error.msg : "",
+            value: context.value || ""
+        };
 
-            //Finds the line number for the start of the error
-            for(i = 0; i <= (msg[1] + 1); i++)
-            {
-                if(input[i] === "\n")
-                {
-                    lineCount += 1;
-                    lineBeginStart = i;
-                }
-                parsedText += input[i];
-                charCount++;
-            }
-            //line number for start of error
-            msg[3] = lineCount;
+        return html
+                .replace(/\[\[closeTag\.name\]\]/, replaceText.closeTag)
+                .replace(/\[\[openTag\.name\]\]/, replaceText.openTag)
+                .replace(/\[\[attribute\.name\.value\]\]/, replaceText.attribute)
+                .replace(/\[\[cssValue\.value\]\]/, replaceText.cssValue)
+                .replace(/\[\[cssSelector\.selector\]\]/, replaceText.cssSelector)
+                .replace(/\[\[cssProperty\.property\]\]/, replaceText.cssProperty)
+                .replace(/\[\[cssKeyword\.value\]\]/, replaceText.cssKeyword)
+                .replace(/\[\[name\]\]/, replaceText.name)
+                .replace(/\[\[error\.msg\]\]/, replaceText.error)
+                .replace(/\[\[value\]\]/, replaceText.value);
+    }
 
-            ////Finds the line number for the end of the error
-            for(i = (msg[1] + 1); i <= (msg[2] + 1); i++)
-            {
-                if(input[i] === "\n")
-                {
-                    lineCount += 1;
-                    lineBeginEnd = i;
-                }
-                parsedText += input[i];
-                charCount++;
-            }
-            //character relative to start of the line
-            msg[1] = msg[1] - lineBeginStart;
-            msg[2] = msg[2] - lineBeginEnd;
-            
-            //line number for end of error
-            msg[4] = lineCount;
+    function parse(input) {
+        var parse = require("slowparse/slowparse");
+        var errorMessages = require("text!slowparse/locale/en_US.json");
```

Comment:
Let's move these requires outside the function call, up near the top of this module.